### PR TITLE
Add support to install ocp on google cloud

### PR DESCRIPTION
**Pull request by chaitanyaenr**
This commit:
- Adds install role to install OCP 4.x on GCP.
- Adds post install roles to create infra, workload nodes and configure
  the cluster to be able to run workloads.
- Adds a readme with the instructions.

### CODE REVIEW

**Comment by chaitanyaenr** [comment]
This is ready for review except that the workload node doesn't get an external IP. The cloud tram is taking a look at it. We need the external ip for debugging purposes but it won't block running the workloads. Will create another PR once the issue is resolved. Thanks.

**Comment by chaitanyaenr** [comment]
The workload node will have a public ip now.

File: OCP-4.X/roles/post-install/templates/gcp-workload-node-machineset.yml.j2 [hunk 1]
```diff
@@ -0,0 +1,65 @@
+apiVersion: machine.openshift.io/v1beta1
```
Comment (akrzos) [inline OCP-4.X/roles/post-install/templates/gcp-workload-node-machineset.yml.j2]: The starting structure of this machineset seems to be a bit different than the [aws](https://github.com/openshift-scale/scale-ci-deploy/blob/master/OCP-4.X/roles/post-install/templates/aws-workload-node-machineset.yml.j2) and azure workload machineset yamls.  It looks like those two yamls are actually lists with a single item in them.  I don't know if it makes more sense to follow this yaml or the others, alternatively if this works, we could also make this a task in the backlog to make the yamls resemble and/or share the same structure as much as possible.

File: docs/README.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ The scale-ci-deploy repo is split into two major directories, playbooks for Open
 | ------------------------ | ----------- | -------- | ------- | ---------------------------- |
 | [AWS](ocp4_aws.md)       | Yes         | Yes      | Yes     | Variable `OPENSHIFT_CLEANUP` |
 | [Azure](ocp4_azure.md)   | Yes         | No       | Yes     | Variable `OPENSHIFT_CLEANUP` |
-| GCP                      | No          | No       | No      | No                           |
+| [GCP](ocp4_gcp.md)       | Yes         | No       | No      | Variable `OPENSHIFT_CLEANUP` |
```
[the remark this answers is not in the document]
Comment (chaitanyaenr) [reply docs/README.md]: Ack.

File: OCP-4.X/vars/install-on-gcp.yml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+# Cluster Configuration
+openshift_base_domain: "{{ lookup('env', 'OPENSHIFT_BASE_DOMAIN') }}"
+openshift_cluster_name: "{{ lookup('env', 'OPENSHIFT_CLUSTER_NAME') }}"
+
+openshift_master_count: "{{ lookup('env', 'OPENSHIFT_MASTER_COUNT')|default(3, true) }}"
+openshift_worker_count: "{{ lookup('env', 'OPENSHIFT_WORKER_COUNT')|default(5, true) }}"
+
+openshift_master_vm_size: "{{ lookup('env', 'OPENSHIFT_MASTER_VM_SIZE')|default('n1-standard-4', true) }}"
+openshift_worker_vm_size: "{{ lookup('env', 'OPENSHIFT_WORKER_VM_SIZE')|default('n1-standard-4', true) }}"
```
[the remark this answers is not in the document]
Comment (chaitanyaenr) [reply OCP-4.X/vars/install-on-gcp.yml]: Makes sense @akrzos, changed it to machine type. Thanks.

**Pull request merged**